src/Core/Content/Category/Event/SalesChannelCategoryIdsFetchedEvent.php

```diff
@@ -0,0 +1,107 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Content\Category\Event;
+
+use Shopware\Core\Content\Category\CategoryException;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\Event\ShopwareEvent;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Uuid\Uuid;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+use Symfony\Contracts\EventDispatcher\Event;
+
+/**
+ * Event that is triggered when category ids are fetched for a sales channel without using the DAL.
+ */
+#[Package('discovery')]
+final class SalesChannelCategoryIdsFetchedEvent extends Event implements ShopwareEvent
+{
+    final public const NAME = 'sales_channel.category.keys.fetched';
+
+    /**
+     * @var array<string, string>
+     */
+    private array $filteredIds = [];
```

Comment:
I still don't see, why we need that second property

Reply:
you are not using it by yourself, so I would stick with the most simply version. if there is a need, it would be no problem to provide that. 

but we always need to have in mind, that every public method is additional API we need to support.